proposals/lip-0053.md

```diff
@@ -738,14 +753,14 @@ The Lisk consensus mechanism is designed to create and publish certificates regu
 Setup to gather the required mainchain information:
 
 * Have access to the Interoperability module endpoints exposed by a mainchain node.
-* Maintain a list of all CCMs included in the sidechain outbox. For each height where a CCM was included in the outbox, also save the inclusion witness of the outbox root with respect to the state root. All CCMs and witnesses for heights that have been certified on the sidechain can be discarded.
-* Maintain a history of all validator changes on the mainchain for rounds that have not yet been certified on the sidechain.
+* Maintain a list of all CCMs included in the sidechain outbox. For each height where a CCM was included in the outbox, also save the inclusion witness of the outbox root with respect to the state root. CCMs can be discarded once they are included via a CCU in a final block of the sidechain. All witnesses corresponding to outbox root values older than the latest `partnerChainOutboxRoot` in the final part of the sidechain can be discarded as well.
+* Maintain a history of all validator changes on the mainchain for rounds that have not yet been certified by a certificate that is included in the final part of the sidechain.
 
 Create a cross-chain update transaction for a given height `h1`:
 
 * Find a [signed certificate][lip-0061#select-certificate] in the mainchain block headers for a height (say `h2`) higher or equal to `h1`. This will be used as the `certificate` property of the transaction.
 * The property `inboxUpdate.crossChainMessages` lists a subset of the CCMs that have been included in the sidechain outbox up to `h2` (and which have not been included on the sidechain yet). If all CCMs are included, the `messageWitnessHashes` will be empty.
-* Compute the inclusion proof for the outbox root of the sidechain account with respect to the mainchain state root. This proof is then used to compute `inboxUpdate.outboxRootWitness`.
+* Select from the list of stored inclusion proofs the one for height `h2`. This proof is then used to compute `inboxUpdate.outboxRootWitness`.
 * Compute the required update to the active validators stored in the chain account and the validators that were used to create `certificate.validatorsHash`. This update can be obtained by following the logic of `getActiveValidatorsUpdate` as detailed in the [Appendix](#appendix).
```

Comment:
Fix link (should link to section "computing the validators update", not appendix, since we are already there).